### PR TITLE
[SQL] Add cmdlet to refresh external governance status.

### PR DESCRIPTION
**Pull request by guptanitish-ms**
## Description

Add cmdlet (Refresh-AzSqlServerExternalGovernanceStatus) to refresh external governance status.
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Review by krivi37 — COMMENTED** [review]
Help files also need to be added. You can follow: https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/help-generation.md

File: src/Sql/Sql/Auditing/Services/AuditingEndpointsCommunicator.cs
[the remark this answers is not in the document]
Comment (guptanitish-ms) [reply src/Sql/Sql/Auditing/Services/AuditingEndpointsCommunicator.cs:1]: removed.

File: src/Sql/Sql/ExternalGovernance/Cmdlet/RefreshAzSqlServerExternalGovernanceStatus.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+﻿using Microsoft.Azure.Commands.Sql.ExternalGovernance.Model;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.ExternalGovernance.Cmdlet
+{
+    [Cmdlet("Refresh", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlServerExternalGovernanceStatus", SupportsShouldProcess = true), OutputType(typeof(RefreshExternalGovernanceModel))]
```
[the remark this answers is not in the document]
Comment (guptanitish-ms) [reply src/Sql/Sql/ExternalGovernance/Cmdlet/RefreshAzSqlServerExternalGovernanceStatus.cs:8]: Lets sync offline.

**Comment by guptanitish-ms** [comment]
> Help files also need to be added. You can follow: https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/help-generation.md

yeah will add. Need to fill the help, etc.

File: src/Sql/Sql/ExternalGovernance/Cmdlet/RefreshAzSqlServerExternalGovernanceStatus.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (guptanitish-ms) [reply src/Sql/Sql/ExternalGovernance/Cmdlet/RefreshAzSqlServerExternalGovernanceStatus.cs:8]: Synced offline. I know refresh is not in list of allowed verbs but the REST API is called RefreshExternalGovernance status. Hence, I would like to keep the PS cmdlet same name. 

Refresh semantically makes a lot of sense for this API.

**Comment by guptanitish-ms** [comment]
> Help files also need to be added. You can follow: https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/help-generation.md
added.

**Comment by guptanitish-ms** [comment]
> Help files also need to be added. You can follow: https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/help-generation.md

added

**Comment by guptanitish-ms** [comment]
> Help files also need to be added. You can follow: https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/help-generation.md

added

File: src/Sql/Sql/Az.Sql.psd1 [hunk 2]
```diff
@@ -53,7 +53,7 @@ DotNetFrameworkVersion = '4.7.2'
 # ProcessorArchitecture = ''
 
 # Modules that must be imported into the global environment prior to importing this module
-RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '2.12.0'; })
+#RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '2.12.0'; })
```
Comment (VeryEarly) [inline src/Sql/Sql/Az.Sql.psd1:56]: please do not comment this line

File: src/Sql/Sql/ChangeLog.md [hunk 3]
```diff
@@ -19,6 +19,10 @@
 -->
 ## Upcoming Release
 
+## Version 4.5.0
```
Comment (VeryEarly) [inline src/Sql/Sql/ChangeLog.md:22]: please move release notes under line 20

File: src/Sql/Sql/help/Refresh-AzSqlServerExternalGovernanceStatus.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.Sql.dll-Help.xml
+Module Name: Az.Sql
+online version:
```
Comment (VeryEarly) [inline src/Sql/Sql/help/Refresh-AzSqlServerExternalGovernanceStatus.md:4]: https://learn.microsoft.com/powershell/module/az.sql/refresh-azsqlserverexternalgovernancestatus

Comment (guptanitish-ms) [reply src/Sql/Sql/help/Refresh-AzSqlServerExternalGovernanceStatus.md:4]: The link is giving 404. Is that the issue?

File: src/Sql/Sql/Az.Sql.psd1 (hunk 2 above)
Comment (guptanitish-ms) [reply src/Sql/Sql/Az.Sql.psd1]: removed.

File: src/Sql/Sql/ExternalGovernance/Cmdlet/RefreshAzSqlServerExternalGovernanceStatus.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (guptanitish-ms) [reply src/Sql/Sql/ExternalGovernance/Cmdlet/RefreshAzSqlServerExternalGovernanceStatus.cs]: Renamed.

**Review by viparek — DISMISSED** [review]

**Review by krivi37 — DISMISSED** [review]

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**